File: Java/70. Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n == 0) return 0;
        int[] steps = new int[n + 1];
        steps[0] = steps[1] = 1;
        for (int i = 2; i < n + 1; i++)
            steps[i] = steps[i - 2] + steps[i - 1];        
        return steps[n];        
    }
}

File: 2021/70. Climbing Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int dp[2];
        dp[0] = dp[1] = 1;
        int updateIndex = 0;
        for (int i = 2; i <= n; i++) {
            int tmp = dp[0] + dp[1];
            dp[updateIndex] = tmp;
            updateIndex = 1 - updateIndex;
        }
        return dp[1 - updateIndex];
    }
};

class Solution {
public:
    int climbStairs(int n) {
        int prev1 = 1, prev2 = 0;
        for (int i = 1; i <= n; i++) {
            int curr = prev1 + prev2;
            prev2 = prev1;
            prev1 = curr;
        }
        return prev1;
    }
};

class Solution {
public:
    int climbStairs(int n) {
        int prev2 = 0, prev1 = 1;
        for (int i = 1; i <= n; i++) {
            int curr = prev2 + prev1;
            prev2 = prev1;
            prev1 = curr;
        }
        return prev1;
    }